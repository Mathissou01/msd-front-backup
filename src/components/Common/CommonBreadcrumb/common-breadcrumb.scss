@use "src/styles/01-settings/settings.colors" as colors;
@use "src/styles/02-tools/tools.media-queries" as media;
@use "src/styles/02-tools/tools.buttons" as buttons;
@use "src/styles/02-tools/tools.typography" as typo;
@use "src/styles/02-tools/tools.functions" as func;

.c-CommonBreadcrumb {
  display: flex;
  gap: 8px;
  align-items: baseline;
  padding: 16px;

  &__Link {
    @include typo.text("buttonSmall");

    font-size: 12px;
    text-transform: capitalize;

    @include media.from-tablet {
      font-size: 14px;
    }

    &_underline {
      text-decoration: underline;
    }
  }
}
