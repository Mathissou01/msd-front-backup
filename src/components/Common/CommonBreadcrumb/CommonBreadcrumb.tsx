import React from "react";
import { useRouter } from "next/router";
import Link from "next/link";
import Breadcrumb from "public/images/pictos/breadcrumb-separator.svg";
import "./common-breadcrumb.scss";

export default function CommonBreadcrumb() {
  /* Local Data */
  const router = useRouter();
  const pathLevelsArray = router.pathname.split("/").slice(1);
  let progressivePath = "";
  const homePage = "/";
  /* Method */
  function handlePathgeneration() {
    return pathLevelsArray.map((path, index) => {
      progressivePath = `${progressivePath}/${path}`;

      return (
        <>
          {router.pathname !== "/" ? (
            <Breadcrumb data-testid="breadcrumb-separator" />
          ) : null}
          <Link
            key={path}
            className={
              pathLevelsArray.length - 1 > index
                ? "c-CommonBreadcrumb__Link c-CommonBreadcrumb__Link_underline"
                : "c-CommonBreadcrumb__Link"
            }
            href={`${progressivePath}`}
          >
            {path}
          </Link>
        </>
      );
    });
  }

  return (
    <nav className="c-CommonBreadcrumb">
      {router.pathname !== homePage ? (
        <Link
          className="c-CommonBreadcrumb__Link c-CommonBreadcrumb__Link_underline"
          href={homePage}
        >
          Accueil
        </Link>
      ) : null}
      {handlePathgeneration()}
    </nav>
  );
}
