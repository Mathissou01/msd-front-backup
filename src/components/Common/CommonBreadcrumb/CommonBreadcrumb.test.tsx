import { render, screen } from "@testing-library/react";
import CommonBreadcrumb from "./CommonBreadcrumb";

const mockPagesUrl = [
  {
    label: "Accueil",
    slug: "/",
  },
  {
    label: "Actualités et événements",
    slug: "actualites-evenements",
  },
  {
    label: "Semaine initiatives vertes",
  },
];

describe("CommonBreadcrumb", () => {
  it("renders", () => {
    const { container } = render(<CommonBreadcrumb pages={mockPagesUrl} />);
    const lableUrl = screen.getByText("Accueil");
    expect(lableUrl).toHaveAttribute("href", "/");
    expect(container).toMatchSnapshot();
  });
});
