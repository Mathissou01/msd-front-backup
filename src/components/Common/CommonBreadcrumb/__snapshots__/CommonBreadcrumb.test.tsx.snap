// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonBreadcrumb renders 1`] = `
<div>
  <nav
    class="c-CommonBreadcrumb"
    role="navigation"
  >
    <a
      class="c-CommonBreadcrumb__Link c-CommonBreadcrumb__Link_active"
      href="/"
    >
      Accueil
    </a>
    <a
      class="c-CommonBreadcrumb__Link c-CommonBreadcrumb__Link_active"
      href="actualites"
    >
      Actualit√©s
    </a>
    <span
      class="c-CommonBreadcrumb__Link"
    >
      Semaine initiatives vertes
    </span>
  </nav>
</div>
`;
